File: app/src/main/java/com/chatimmi/usermainfragment/connectfragment/Chats.java
package com.chatimmi.usermainfragment.connectfragment;

import android.net.Uri;

import java.io.Serializable;

public class Chats implements Serializable {
    //Todo Socket chat used variable
    public String name;
    public String  uid;
    public String chatRoomId;
    public String channelId;
    public String title;
    public String roomType;
    public String messageId;
    public String fileType="";
    public String imageUrl="";
    public String currentTime="";
    public String createdTime="";
    public String from="";
    public int unreadCount;
    public int badgeCount;
    public int isOnline;
    public Long lastSeenTimeStamp;
    public int isNotification;
    public int muteNotification;
    public int memberType;
    public int member_type_reference_id;
    public String member_tlastSeenTimeStampype_reference_id;
    public String eventEndDate;


    //Todo firebase chat used variable
    public String deleteby;
    public String firebaseId;
    public String firebaseToken;
    public String authToken;
    public int   image;
    public String message;
    public String profilePic;
    public Long timestamp;
    public String lastMsg;
    public String lastMsgTime;
    public String isBlocked;
    public String socketRoom;
    public String lastMsgId;
    public String lastSeenString;

    public void Chat() {
    }

    public String banner_date="";
    public int isMsgReadTick;
    public Uri imgURI;

    public String type;
    public boolean isGroup=false;
    public int groupUserCount=0;
    public String userName;
    public String message_type;


    public EventInfoDetails eventInfoDetails;


    public static class EventInfoDetails implements Serializable{
        public EventInfoDetails() {
        }

        public String eventId;
        public String ownerType;
        public String eventName;
        public String eventType;
        public String eventImage;
        public String eventOrganizerId;
        public String eventOrganizerName;
        public String eventOrganizerProfileImage;
        public String compId;
        public String eventMemId;

    }
}

File: app/src/main/java/com/chatimmi/usermainfragment/connectfragment/Chats.kt
package com.chatimmi.usermainfragment.connectfragment

import android.net.Uri
import java.io.Serializable



class Chats : Serializable {
    //Todo Socket chat used variable
    var name: String? = null
    var uid: String? = null
    var chatRoomId: String? = null
    var channelId: String? = null
    var title: String? = null
    var roomType: String? = null
    var messageId: String? = null
    var fileType = ""
    var imageUrl = ""
    var currentTime = ""
    var createdTime = ""
    var from = ""
    var unreadCount = 0
    var badgeCount = 0
    var isOnline = 0
    var lastSeenTimeStamp: Long? = null
    var isNotification = 0
    var muteNotification = 0
    var memberType = 0
    var member_type_reference_id = 0
    var member_tlastSeenTimeStampype_reference_id: String? = null
    var eventEndDate: String? = null

    //Todo firebase chat used variable
    var deleteby: String? = null
    var firebaseId: String? = null
    var firebaseToken: String? = null
    var authToken: String? = null
    var image = 0
    var message: String? = null
    var profilePic: String? = null
    var timestamp: Long? = null
    var lastMsg: String? = null
    var lastMsgTime: String? = null
    var isBlocked: String? = null
    var socketRoom: String? = null
    var lastMsgId: String? = null
    var lastSeenString: String? = null
    var banner_date = ""
    var isMsgReadTick = 0
    var imgURI: Uri? = null
    var type: String? = null
    var isGroup = false
    var groupUserCount = 0
    var userName: String? = null
    var message_type: String? = null
    var eventInfoDetails: EventInfoDetails? = null


    class EventInfoDetails : Serializable {
        var eventId: String? = null
        var ownerType: String? = null
        var eventName: String? = null
        var eventType: String? = null
        var eventImage: String? = null
        var eventOrganizerId: String? = null
        var eventOrganizerName: String? = null
        var eventOrganizerProfileImage: String? = null
        var compId: String? = null
        var eventMemId: String? = null
    }
}
